Schema Edits - Asset Types

theHostileNegotiator opened the pull request.

theHostileNegotiator committed:
Schemas Assets

Qibbi commented on SAGE/Games/Kane's Wrath/GameObject.xml:
```diff
@@ -1298,78 +1094,24 @@
-			id="Weapon"
-			AssetType="WeaponTemplate"
+			id="AutoResolveWeapon"
+			AssetType="ArmorTemplate"
```
This should be a WeaponTemplate.
Also shuffling around stuff is really messing with diffing.

theHostileNegotiator replied:
I would like to follow the same ordering as the EA Schemas. Perhaps I make 2 commits, the first one will be any edits and additions to all Asset Types and the second one just reshuffles the GameObject file

Qibbi replied:
If you mean 2 prs then yes, that'd be great, as it'd be a lot easier to compare.
But as I compared it already anyway just fix the asset type issue above and I'll merge it.

theHostileNegotiator replied:
I've unshuffled it and fixed the issue

Qibbi commented on SAGE/Games/Kane's Wrath/TerrainTextureAtlas.xml:
```diff
@@ -88,6 +88,10 @@
 			AssetType="SageBool"
 			IsAttribute="true"
 			Default="true" />
+		<Entry
+			id="fill"
+			AssetType="Byte"
+			IsVoid="true" />
```
Not necessary for types which have a different runtime type.

Qibbi commented on SAGE/Games/Kane's Wrath/Texture.xml:
```diff
@@ -49,6 +49,10 @@
 			AssetType="SageBool"
 			IsAttribute="true"
 			Default="true" />
+		<Entry
+			id="fill"
+			AssetType="Byte"
+			IsVoid="true" />
```
Not necessary for types which have a different runtime type.

Qibbi commented on SAGE/Games/Kane's Wrath/Texture.xml:
```diff
@@ -86,6 +90,10 @@
 			AssetType="SageBool"
 			IsAttribute="true"
 			Default="true" />
+		<Entry
+			id="fill"
+			AssetType="Byte"
+			IsVoid="true" />
```
Not necessary for types which have a different runtime type.

Qibbi commented on SAGE/Games/Kane's Wrath/W3D.xml:
```diff
@@ -343,6 +343,10 @@
 			id="UsageIndex"
 			AssetType="Byte"
 			IsAttribute="true" />
+		<Entry
+			id="fill"
+			AssetType="Byte"
+			IsVoid="true" />
```
Nothing to fill as it's already aligned on the 4 byte boundary.

Qibbi commented on SAGE/Games/Kane's Wrath/W3D.xml:
```diff
@@ -372,6 +376,10 @@
 			AssetType="SageUnsignedShort"
 			MinLength="1"
 			MaxLength="0" />
+		<Entry
+			id="fill"
+			AssetType="Byte"
+			IsVoid="true" />
```
This shouldn't be here.

Qibbi reviewed (changes requested):
There are some issues, but great effort.
If you could un-shuffle the gameobject.xml would be great.

theHostileNegotiator commented:
I've amended the Commit to address the unnecessary fills for runtime types, unshuffled GameObject.xml, fixed existing error and added project includes for new files
Before merging, let me know if you approve the final changes, and I'll add a commit to shuffle the GameObject file, then you can review and Merge

karim20027 commented:
i have redalert3 assettypes with all right typehashes and all working except gameobject and weapontemplate can i send to you the files and can you fix them?
[redalert3.zip](https://github.com/user-attachments/files/27570524/redalert3.zip)
